File: python/parse_decode_image_task.py
import argparse

import decode_image_task

parser = argparse.ArgumentParser()
parser.add_argument('file_path', help='file path')
parser.add_argument('show_undone_part_num', type=int, help='show undone part num')
args = parser.parse_args()

task = decode_image_task.Task(args.file_path)
task.load()
task.print(args.show_undone_part_num)

File: src/parse_decode_image_task/parse_decode_image_task.cpp
#include <iostream>
#include <string>

#include "image_codec.h"

int main(int argc, char** argv)
{
    if (argc != 3) {
        std::cout << "parse_decode_image_task file_path show_undone_part_num";
        return 1;
    }
    Task task(argv[1]);
    task.Load();
    task.Print(std::stoll(argv[2]));

    return 0;
}